### PR TITLE
Dind audit policy fixes #9060

### LINKED ISSUE #9060: Add audit logging / policy to kubetest + dind
Audit logging is quite useful.

kubetest dind / calls kubeadm without an audit-policy or logging in place.

I would like to enable kubetest / dind / audit-logging.

### PR DESCRIPTION
**Pull request by hh**

### CODE REVIEW

File: kubetest/dind/dind.go [hunk 1]
```diff
@@ -98,6 +98,7 @@ func (d *Deployer) BuildTester(o *e2e.BuildTesterOptions) (e2e.Tester, error) {
 	// dind tester sets parallelism a little differently
 	t.GinkgoParallel = 1
 	t.NumNodes = o.Parallelism
+	t.Provider = "skeleton"
```
Comment (hh) [inline kubetest/dind/dind.go]: @BenTheElder should we set this statically? I can't imagine someone using a value other than Skeleton

Comment (BenTheElder) [reply kubetest/dind/dind.go]: I think this makes sense. I'll try to follow up for something more thorough when kind integrates with kubetest

File: dind/start.sh [hunk 2]
```diff
@@ -95,8 +95,45 @@ start_master ()
...
+  resources:
+  - group: "" # core
+    resources: ["pods", "secrets"]
+  - group: "extensions"
+    resources: ["deployments"]
+EOPOLICY
+  chmod 0600 /etc/kubernetes/audit-policy.yaml
+  cat <<EOF > /etc/kubernetes/kubeadm.conf
+#apiServerExtraArgs:
+#  cert-extra-sans: $1
```
Comment (BenTheElder) [inline dind/start.sh]: any reason for these commented lines?

File: dind/start.sh [hunk 3]
```diff
@@ -95,8 +95,45 @@ start_master ()
...
+  path: /etc/kubernetes/audit-policy.yaml
+featureGates:
+  Auditing: true
+networking:
+  podSubnet: 192.168.0.0/16
+bootstrapTokens:
+- groups:
+  - system:bootstrappers:kubeadm:default-node-token
+  token: abcdef.abcdefghijklmnop
+  #token: tzc9vo.ugspyk7o3jxxhak6
```
Comment (BenTheElder) [inline dind/start.sh]: likewise, probably we should drop this line?

**Comment by BenTheElder** [comment]
Generally LGTM, two nits.

File: dind/start.sh (hunk 2 above)
Comment (hh) [reply dind/start.sh]: I'm still working on passing cert-extra-sans... the fileformat and command line args have changed... still hunting down the v1alpha3 format.

File: dind/node/Dockerfile [hunk 4]
```diff
@@ -41,7 +41,11 @@ RUN mkdir -p /opt/cni/bin
 RUN dpkg -i kubectl.deb
 RUN dpkg -i kubernetes-cni.deb
 RUN dpkg -i kubelet.deb
-RUN dpkg -i kubeadm.deb
+ENV CRI_VERSION="v1.11.1"
+RUN curl -L https://github.com/kubernetes-incubator/cri-tools/releases/download/$CRI_VERSION/crictl-$CRI_VERSION-linux-amd64.tar.gz | \
+    tar zxvf - -C /usr/local/bin
+# force due to dependency on missing cri deb for now
+RUN dpkg -i --force depends kubeadm.deb
```
Comment (BenTheElder) [inline dind/node/Dockerfile]: there's a CRI deb in k/k build we can consume https://github.com/kubernetes/kubernetes/commit/03456a33f9cd1123572a4a20e3728b65d5fb6925

File: dind/start.sh [hunk 5]
```diff
@@ -95,8 +95,44 @@ start_master ()
...
+auditPolicy:
+  path: /etc/kubernetes/audit-policy.yaml
+featureGates:
+  Auditing: true
+networking:
+  podSubnet: 192.168.0.0/16
+bootstrapTokens:
+- groups:
+  - system:bootstrappers:kubeadm:default-node-token
+  token: abcdef.abcdefghijklmnop
```
Comment (hh) [inline dind/start.sh]: This does result it:

```
You can now join any number of machines by running the following on each node
as root:

  kubeadm join 172.18.0.2:6443 --token abcdef.abcdefghijklmnop --discovery-token-ca-cert-hash sha256:008789ee5ec6758715f39fda15406615c0d7150eb386e5b794cdd066640d46a2
```

However on my minion nodes:

```
[discovery] Created cluster-info discovery client, requesting info from "https://172.18.0.2:6443"
[discovery] Failed to connect to API Server "172.18.0.2:6443": token id "abcdef" is invalid for this cluster or it has expired. Use "kubeadm token create" on the master node to creating a new valid token
[discovery] abort connecting to API servers after timeout of 5m0s
```
I'm unsure if this is related to the certs:

```
E0816 23:30:21.344248       1 memcache.go:134] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
I0816 23:30:31.338437       1 logs.go:50] http: TLS handshake error from 172.17.0.1:59826: remote error: tls: bad certificate
I0816 23:30:41.339545       1 logs.go:50] http: TLS handshake error from 172.17.0.1:59838: remote error: tls: bad certificate
I0816 23:30:51.335838       1 logs.go:50] http: TLS handshake error from 172.17.0.1:59850: remote error: tls: bad certificate
```

File: dind/start.sh (hunk 2 above)
Comment (hh) [reply dind/start.sh]: This is fixed now, it had to be a list... and 'kubernetes' also needed to be added

File: dind/start.sh (hunk 3 above)
Comment (hh) [reply dind/start.sh]: Done

File: dind/node/Dockerfile (hunk 4 above)
Comment (hh) [reply dind/node/Dockerfile]: Updated to include! Thanks for helping me find that!

File: dind/start.sh (hunk 5 above)
Comment (hh) [reply dind/start.sh]: This issue went away when we added apiServerCertSANs in this way:

```yaml
apiServerCertSANs:
$(echo $1 | sed -e 's: :\n:g' | sed 's:^:- :') 
- kubernetes 
```

**Comment by spiffxp** [comment]
@hh do you still want to land this? it looks like one of the commits is authored as "root" and the CNCF CLA bot isn't happy with that

**Comment by BenTheElder** [comment]
FWIW I'd love to get some fixes in for this to [sigs.k8s.io/kind](https://sigs.k8s.io/kind)

**Comment by BenTheElder** [comment]
dind/ has been removed. Would still love to get audit logging working nice for [sigs.k8s.io/kind](https://sigs.k8s.io/kind), just haven't had the time yet.

**Pull request closed without merging**